packages/hierarchies-react/src/test/UseSelectionHandler.test.tsx

```diff
@@ -0,0 +1,417 @@
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1", "child-outer", "child-inner", "node-2"], "replace");
+      });
+
+      it("skips non visible children when selecting range", async () => {
+        const innerChild = createHierarchyNode("child-inner");
+        const outerChild = createHierarchyNode("child-outer", [innerChild]);
+        const nodes = [createHierarchyNode("node-1", [outerChild], false), createHierarchyNode("node-2")];
+        const { user, getByText } = render(<TestComponent {...createProps(nodes, selectionMode, true)} />);
+
+        const node1 = getByText("node-1");
+        node1.focus();
+        await clickNode(user, node1);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1"], "replace");
+        selectNodesStub.reset();
+
+        const node2 = getByText("node-2");
+        node2.focus();
+
+        await user.keyboard(`{Shift>}`);
+        await clickNode(user, node2);
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1", "node-2"], "replace");
+      });
+
+      it("does not update previous selection when using `shift`", async () => {
+        const nodes = [createHierarchyNode("node-1"), createHierarchyNode("node-2"), createHierarchyNode("node-3"), createHierarchyNode("node-4")];
+        const { user, getByText } = render(<TestComponent {...createProps(nodes, selectionMode, true)} />);
+
+        let node = getByText("node-2");
+        node.focus();
+        await clickNode(user, node);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-2"], "replace");
+        selectNodesStub.reset();
+
+        node = getByText("node-1");
+        node.focus();
+
+        await user.keyboard(`{Shift>}`);
+        await clickNode(user, node);
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-1", "node-2"], "replace");
+        selectNodesStub.reset();
+
+        node = getByText("node-3");
+        node.focus();
+
+        await user.keyboard(`{Shift>}`);
+        await clickNode(user, node);
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-2", "node-3"], "replace");
+        selectNodesStub.reset();
+
+        node = getByText("node-4");
+        node.focus();
+
+        await user.keyboard(`{Control>}`);
+        await clickNode(user, node);
+        await user.keyboard(`{/Control}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-4"], "add");
+        selectNodesStub.reset();
+
+        node = getByText("node-3");
+        node.focus();
+
+        await user.keyboard(`{Shift>}`);
+        await clickNode(user, node);
+        await user.keyboard(`{/Shift}`);
+
+        expect(selectNodesStub).to.be.calledOnceWith(["node-3", "node-4"], "replace");
+      });
+
+      it("clears selection when invalid node passed", async () => {
```

Comment:
We probably should ignore if there is something `onNodeClicked` is called with invalid node.

Reply:
Changed to do nothing if invalid node is selected.